CHANGELOG.next.asciidoc

```diff
@@ -124,6 +124,11 @@ https://github.com/elastic/beats/compare/v8.7.1\...main[Check the HEAD diff]
 - Fix crash when loading azurewebstorage cursor with no partially processed data. {pull}35433[35433]
 - RFC5424 syslog timestamps with offset 'Z' will be treated as UTC rather than using the default timezone. {pull}35360[35360]
 - [system] sync system/auth dataset with system integration 1.29.0. {pull}35581[35581]
+- [GCS Input] - Fixed an issue where bucket_timeout was being applied to the entire bucket poll interval and not individual 
+   bucket object read operations. Fixed a map write concurrency issue arising from data races when using a high number of workers. 
+   Fixed the flaky tests that were present in the GCS test suit. {pull}35605[35605]
+- [GCS Input] - Enhanced the input by adding debug logs and improved the context passing mechanism by removing them from struct params 
+   and passing them as function arguments. {pull}35605[35605]
```

Comment:
I'd like the bug fixes to be in the same PR so that they can be fast tracked as they relate to an urgent SDH. I can separate out the enhancements into a separate PR(context and logging).